smoker/src/smoker/backends/dummybackend.go

```diff
@@ -344,47 +342,49 @@ func (b *DummyBackend) AddComponent(comp Component) (Bin, error) {
 	}
 
 	// Actually add component to bin
-	selectedBin.parts[comp] = true
-	comp.setBin(selectedBin)
+	selectedBin.Parts[comp.GetId()] = true
+	comp.setBin(selectedBin.GetName())
 
 	// Add lookup pointers for us
-	b.idLookup[comp.GetId()] = comp
-	b.components[comp] = true
+	b.IdLookup[comp.GetId()] = comp
 	return selectedBin, nil
 }
 
 // Moves a component from it's current bin to a valid one
 func (b *DummyBackend) MoveComponent(comp Component, name string) error {
-	if comp.GetBin() == nil {
+	if comp.GetBin() == "" {
 		return errors.New("Comp is not stored in a bin yet!")
 	}
-	if comp.GetBin().GetName() == name {
+	if comp.GetBin() == name {
 		// We are already in the target bin
 		return nil
 	}
 
-	for _, bin := range b.bins {
+	for _, bin := range b.Bins {
 		if bin.GetName() == name {
-			if bin.GetCapacity() > uint(len(bin.parts)) {
-				comp.GetBin().deletePart(comp)
-				comp.setBin(&bin)
-				bin.parts[comp] = true
+			if bin.GetCapacity() > uint(len(bin.Parts)) {
+				oldbin := b.Bins[comp.GetBin()]
+				oldbin.deletePart(comp.GetId())
+				comp.setBin(bin.GetName())
+				bin.Parts[comp.GetId()] = true
 				// Don't need to touch b.components or b.idLookup
 				return nil
 			}
-			return errors.New("'" + bin.name + "' is over capacity!")
+			return errors.New("'" + bin.GetName() + "' is over capacity!")
 		}
 	}
 	return errors.New("Bin '" + name + "' was not found!")
 }
 func (b *DummyBackend) LookupId(id uint) (Component, Bin, error) {
-	if component, present := b.idLookup[id]; !present {
+	if component, present := b.IdLookup[id]; !present {
 		return nil, nil, errors.New("No component found with that ID.")
 	} else {
-		if component.GetBin() == nil {
+		if component.GetBin() == "" {
 			return nil, nil, errors.New("[INTERNAL] The component found has no bin associated with it.")
 		}
-		return component, component.GetBin(), nil
+		bin := b.Bins[component.GetBin()]
+		// TODO take a look at this logic
+		return component, &bin, nil
```

Comment:
returning `&bin` needs a second look.

Reply:
This needs another refactor, *sigh*.

Reply:
Seems ok, alternative is to store `Bin` objects rather than `DummyBin` objects, but that seems like extra work for no gain.